crates/bevy_ecs/src/system/system_chaining.rs

```diff
@@ -142,3 +142,131 @@ where
         }
     }
 }
+
+/// A collection of common adapters for [chaining](super::ChainSystem) the result of a system.
+pub mod adapter {
+    use crate::system::In;
+    use std::fmt::Debug;
+
+    /// Converts a regular function into a system adapter.
+    ///
+    /// # Examples
+    /// ```
+    /// use bevy_ecs::prelude::*;
+    ///
+    /// return1
+    ///     .chain(system_adapter::new(u32::try_from))
+    ///     .chain(system_adapter::unwrap)
+    ///     .chain(print);
+    ///
+    /// fn return1() -> u64 { 1 }
+    /// fn print(In(x): In<impl std::fmt::Debug>) {
+    ///     println!("{x:?}");
+    /// }
+    /// ```
+    pub fn new<T, U>(mut f: impl FnMut(T) -> U) -> impl FnMut(In<T>) -> U {
```

Comment:
IMHO `new` may be confusing. In rust `new` is generally used to create an object.
Since this is a mapping operation, I think `map` would fit better.

Reply:
I respectfully disagree. The meaning behind the `system_adapter::new` is "create a new system adapter from some fn", which should be clear enough IMO.

I don't think it makes sense to call it `map`, since the mapping is semantically being done by the `.chain()` call.

Reply:
I didn't take into consideration the mod name. It makes sense.